packages/core/core-flows/src/order/workflows/exchange-add-new-item.ts

```diff
@@ -0,0 +1,109 @@
+import {
+  OrderChangeDTO,
+  OrderDTO,
+  OrderExchangeDTO,
+  OrderWorkflow,
+} from "@medusajs/types"
+import { ChangeActionType } from "@medusajs/utils"
+import {
+  WorkflowData,
+  createStep,
+  createWorkflow,
+  transform,
+} from "@medusajs/workflows-sdk"
+import { useRemoteQueryStep } from "../../common"
+import { createOrderChangeActionsStep } from "../steps/create-order-change-actions"
+import { previewOrderChangeStep } from "../steps/preview-order-change"
+import {
+  throwIfIsCancelled,
+  throwIfOrderChangeIsNotActive,
+  throwIfOrderIsCancelled,
+} from "../utils/order-validation"
+import { addOrderLineItemsWorkflow } from "./add-line-items"
+
+const validationStep = createStep(
+  "exchange-add-new-item-validation",
+  async function ({
+    order,
+    orderChange,
+    orderExchange,
+  }: {
+    order: OrderDTO
+    orderExchange: OrderExchangeDTO
+    orderChange: OrderChangeDTO
+  }) {
+    throwIfOrderIsCancelled({ order })
+    throwIfIsCancelled(orderExchange, "Exchange")
+    throwIfOrderChangeIsNotActive({ orderChange })
+  }
+)
+
+export const orderExchangeAddNewItemWorkflowId = "exchange-add-new-item"
+export const orderExchangeAddNewItemWorkflow = createWorkflow(
```

Comment:
thought(non-blocking): It almost looks like we might be able to create a generic step for adding the line item action changes. The steps for the different "source" (order, exchange, claim) models are very similar.

Reply:
They will be similar indeed. I can try to come up with something generic later, or at least a portion of it.